src/test/java/com/microsoft/sqlserver/jdbc/resiliency/BasicConnectionTest.java

```diff
@@ -0,0 +1,266 @@
+    @Tag(Constants.xAzureSQLDW)
+    public void testPooledConnectionDB() throws SQLException {
+        SQLServerConnectionPoolDataSource mds = new SQLServerConnectionPoolDataSource();
+        mds.setURL(connectionString);
+        PooledConnection pooledConnection = mds.getPooledConnection();
+        String newDBName = null;
+        String resultDBName = null;
+        String originalDBName = null;
+        try (Connection c = pooledConnection.getConnection(); Statement s = c.createStatement()) {
+            ResiliencyUtils.minimizeIdleNetworkTracker(c);
+            ResultSet rs = s.executeQuery("SELECT DB_NAME();");
+            rs.next();
+            originalDBName = rs.getString(1);
+            newDBName = RandomUtil.getIdentifier("resDB");
+            TestUtils.dropDatabaseIfExists(newDBName, connectionString);
+            s.execute("CREATE DATABASE [" + newDBName + "]");
+            try {
+                s.execute("USE [" + newDBName + "]");
+            } catch (SQLException e) {
+                return;
+            }
+            c.close();
+            Connection c1 = pooledConnection.getConnection();
+            Statement s1 = c1.createStatement();
+            ResiliencyUtils.killConnection(c1, connectionString);
+            ResiliencyUtils.minimizeIdleNetworkTracker(c1);
+            rs = s1.executeQuery("SELECT db_name();");
+            while (rs.next()) {
+                resultDBName = rs.getString(1);
+            }
+            // Check if the driver reconnected to the expected database.
+            assertEquals(originalDBName, resultDBName);
+        } finally {
+            TestUtils.dropDatabaseIfExists(newDBName, connectionString);
+        }
+    }
+
+    @Test
+    public void testPooledConnectionLang() throws SQLException {
+        SQLServerConnectionPoolDataSource mds = new SQLServerConnectionPoolDataSource();
+        mds.setURL(connectionString);
+        PooledConnection pooledConnection = mds.getPooledConnection();
+        String lang0 = null, lang1 = null;
+
+        try (Connection c = pooledConnection.getConnection(); Statement s = c.createStatement()) {
+            ResiliencyUtils.minimizeIdleNetworkTracker(c);
+            ResultSet rs = s.executeQuery("SELECT @@LANGUAGE;");
+            while (rs.next())
+                lang0 = rs.getString(1);
+            s.execute("SET LANGUAGE FRENCH;");
+            c.close();
+            try (Connection c1 = pooledConnection.getConnection(); Statement s1 = c1.createStatement()) {
+                ResiliencyUtils.killConnection(c1, connectionString);
+                ResiliencyUtils.minimizeIdleNetworkTracker(c1);
+                rs = s1.executeQuery("SELECT @@LANGUAGE;");
+                while (rs.next())
+                    lang1 = rs.getString(1);
+                assertEquals(lang0, lang1);
+                s1.close();
+                c1.close();
+            } finally {
+                rs.close();

```

Comment:
could you close all in finally?
rs > s1 > c1

Reply:
Actually s1 and c1 will get cleaned up automatically, there is no need to call close for them, will change.